airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/MetricQueries.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import static io.airbyte.db.instance.configs.jooq.Tables.ACTOR;
+import static io.airbyte.db.instance.configs.jooq.Tables.ACTOR_DEFINITION;
+
+import io.airbyte.db.instance.configs.jooq.enums.ReleaseStage;
+import java.util.List;
+import java.util.UUID;
+import org.jooq.DSLContext;
+
+/**
+ * Keep track of all metric queries.
+ */
+public class MetricQueries {
+
+  public static List<ReleaseStage> jobIdToReleaseStages(final DSLContext ctx, final long jobId) {
+    final var srcRelStageCol = "src_release_stage";
+    final var dstRelStageCol = "dst_release_stage";
+
+    final var query = String.format("""
+                                    SELECT src_def_data.release_stage AS %s,
+                                           dest_def_data.release_stage AS %s
+                                    FROM connection
+                                    INNER JOIN jobs ON connection.id=CAST(jobs.scope AS uuid)
+                                    INNER JOIN actor AS dest_data ON connection.destination_id = dest_data.id
+                                    INNER JOIN actor_definition AS dest_def_data ON dest_data.actor_definition_id = dest_def_data.id
+                                    INNER JOIN actor AS src_data ON connection.source_id = src_data.id
+                                    INNER JOIN actor_definition AS src_def_data ON src_data.actor_definition_id = src_def_data.id
+                                        WHERE jobs.id = '%d';""", srcRelStageCol, dstRelStageCol, jobId);
```

Comment:
it'd be nice to isolate the "complex" SQL queries in separate resources `.sql` files down the line